docs/root/configuration/http_filters/gzip_filter.rst

```diff
@@ -0,0 +1,47 @@
+.. _config_http_filters_gzip:
+
+Gzip
+====
+Gzip is an HTTP filter which enables Envoy to compress dispatched data
+from an upstream service upon client request. Compression is useful in
+situations where large payloads need to be transmitted without
+compromising the response time.
+
+Configuration
+-------------
+* :ref:`v1 API reference <config_http_filters_gzip_v1>`
+* :ref:`v2 API reference <envoy_api_msg_filter.http.Gzip>`
+
+.. attention::
+
+  Due to a known bug in the underlying Zlib library, window bits with value
+  eight does not work as expected. Therefore 9 is the smallest window size
+  supported by gzip filter at the moment. This issue might be solved in
+  future releases of the library.
+
+How it works
+------------
+When gzip filter is enabled, request and response headers are inspected to
+determine whether or not the content should be compressed. The content is
+compressed and then sent to the client with the appropriate headers if either
+response and request allow.
+
+By *default* compression will be *skipped* when:
+
+- A request does NOT contain *Accept-Encoding* header.
+- A request includes *Accept-Encoding* header, but it does not contain "gzip".
+- A response contains a *Content-Encoding* header.
+- A Response includes a *Cache-Control* header whose value is no-transform.
+- A response includes a *Transfer-Encoding* header whose value is "gzip".
+- A response does not contain a *Content-Type* value that matches one of the
+  default mime-types.
```

Comment:
It's not possible to configure a custom mime-type to be gzipped?

Reply:
Yes, it is. That's why `By *default* compression will be *skipped* when:` is at the top.

Reply:
Sorry I'm still confused.  Let's say I define a content-type "x-application/custom".  How would I get that to be compressed?

Reply:
Hi Joshua, maybe I'm not following your question. If any Mime-Type is specified, a fixed list will take place. This is a modification made based one of your previous reviews. If the system admin wants to specify a custom type, it will need to be configured in a new list, so the default list goes away and a new one takes place e.g.,

custom (if specified):
```
{
      "content_type": [
           "x-application/custom",
           "x-application/custom-1",
           "x-application/custom-2",
           "x-application/custom-3",
           "application/json"
     ]
}
```

default (if any is specified):
```
{
      "content_type": [
           "text/html", 
            "text/plain",
            "text/css",
             "application/javascript",
             "application/json",
             "image/svg+xml",
             "text/xml",
             "application/xhtml+xml"
     ]
}
```
The filter will match whatever is on the list. Does that make sense?

Reply:
Did you mean "if *none* is specified" rather than "if *any* is specified?

Sorry I'm trying to be precise here.  I think actually that you should put the above comment in the documentation.  And in the original doc you wrote:

"A response does not contain a *Content-Type* value that matches one of the default mime-types" and I think maybe what you meant is "A response does not contain a *Content-Type* value that matches one of the *selected* mime-types, which default to [list of mime types].

Does that make sense?

Reply:
Yes, it does! I meant "if none is specified" 🤦‍♂️

Reply:
Great.  WDYT of putting all this clarification in the doc itself?